tests/playwright-test/test-modifiers.spec.ts

```diff
@@ -214,3 +214,74 @@ test('modifier with a function should throw in the test', async ({ runInlineTest
   expect(result.exitCode).toBe(1);
   expect(result.output).toContain('test.skip() with a function can only be called inside describe block');
 });
+
+test('test.skip with worker fixtures only should skip before hooks and tests', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'a.test.ts': `
+      const test = pwt.test.extend({
+        bar: 'bar',
+        foo: [ 'foo', { scope: 'worker' }],
+      });
+      const logs = [];
+      test.beforeEach(() => {
+        console.log('\\n%%beforeEach');
+      });
+      test('passed', () => {
+        console.log('\\n%%passed');
+      });
+      test.describe('suite1', () => {
+        test.skip(({ foo }) => foo === 'foo', 'reason');
+        test.beforeAll(() => {
```

Comment:
is beforeAll supposed to run before or after the modifier. Couldn't this beforeAll modify `foo` to change how its behavior in `skip`.

Reply:
In theory it could, but we run the skip before beforeAll. This way we can avoid hooks entirely when `skip` says so.